File: gestionart_backend_ruben_y_jessica/api/src/main/java/com/gestionart/api/domain/models/Usuario.java
package com.gestionart.api.domain.models;

import com.gestionart.api.domain.enums.Rol;

public class Usuario {
    private Long id;
    private String correoElectronico;
    private String nombre;
    private String imagen;
    private String contrasena;
    private Rol rol;
    public Usuario() {
    }
    public Usuario(Long id, String correoElectronico, String nombre, String imagen, String contrasena, Rol rol) {
        this.id = id;
        this.correoElectronico = correoElectronico;
        this.nombre = nombre;
        this.imagen = imagen;
        this.contrasena = contrasena;
        this.rol = rol;
    }
    public Long getId() {
        return id;
    }
    public void setId(Long id) {
        this.id = id;
    }
    public String getCorreoElectronico() {
        return correoElectronico;
    }
    public void setCorreoElectronico(String correoElectronico) {
        this.correoElectronico = correoElectronico;
    }
    public String getNombre() {
        return nombre;
    }
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    public String getImagen() {
        return imagen;
    }
    public void setImagen(String imagen) {
        this.imagen = imagen;
    }
    public String getContrasena() {
        return contrasena;
    }
    public void setContrasena(String contrasena) {
        this.contrasena = contrasena;
    }
    public Rol getRol() {
        return rol;
    }
    public void setRol(Rol rol) {
        this.rol = rol;
    }

    
}

File: gestionart_frontend_ruben_y_jessica/lib/data/models/Usuario.dart


import 'package:gestionart_frontend_ruben_y_jessica/data/enums/Rol.dart';

class Usuario {
  final int id;
  String correoElectronico;
 String nombre;
 String imagen;
 String contrasena;
 String contrasena2;
 Rol rol;

  Usuario({
    required this.id, 
    required this.correoElectronico, 
    required this.nombre, 
    required this.imagen, 
    required this.contrasena, 
    required this.contrasena2, 
    required this.rol
  });

  int getId() {
    return id;
  }

  String getCorreoElectronico() {
    return correoElectronico;
  }

  String getNombre() {
    return nombre;
  }

  String getImagen() {
    return imagen;
  }

  String getContrasena() {
    return contrasena;
  }

  String getContrasena2() {
    return contrasena2;
  }

  Rol getRol() {
    return rol;
  }


  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['id'] ?? 0,
      correoElectronico: json['correoElectronico'] ?? '',
      nombre: json['nombre'] ?? '',
      imagen: json['imagen'] ?? '',
      contrasena: json['contrasena'] ?? '',
      contrasena2: json['contrasena2'] ?? '',
     rol: Rol.values.firstWhere(
        (e) => e.toString().split('.').last == (json['rol'] ?? 'COMPRADOR'),
        orElse: () => Rol.COMPRADOR,
      ),
    );
  } 

  Map<String, dynamic> toJson() {
    return {
      'id': id ?? 0,
      'correoElectronico': correoElectronico ?? '',
      'nombre': nombre ?? '',
      'imagen': imagen ?? '',
      'contrasena': contrasena ?? '',
      'contrasena2': contrasena2 ?? '',
      'rol': rol.toString().split('.').last ?? Rol.COMPRADOR.toString(),
    };
  }

}